backend/actions/validate.js

```diff
@@ -0,0 +1,83 @@
+const app = require('../app.js');
+const logger = require('../libs/logger.js');
+const rest = require('../libs/rest');
+const jsontokens = require('jsontokens');
+
+app.use((req, res, next) => {
+  res.header('Access-Control-Allow-Origin', '*');
+  res.header(
+    'Access-Control-Allow-Headers',
+    'Origin, X-Requested-With, Content-Type, Accept'
+  );
+  next();
+});
+
+const isValidCommitment = (script, commitment) => {
+  if (!script || !commitment) {
+    return false;
+  }
+
+  return (
+    script.substring(0, 2) == '6a' && // OP_RETURN
+    script.substring(2, 4) == '26' && // size(38 bytes)
+    script.substring(4, 8) == '5450' && // marker
+    script.substring(8, 10) == '02' && // version
+    script.substring(10, 12) == '22' && // payload size(22 bytes)
+    (script.substring(12, 14) == '01' ||
+      script.substring(12, 14) == '02' ||
+      script.substring(12, 14) == '03') && // operation
+    script.substring(14) == commitment
+  ); //commitment
```

Comment:
Commitments must be calculated from each material and checked to match payload commitments.

Reply:
875285639192fb82dd6ce28b16e1c560c63f8007